wnfs-common/src/utils/test.rs

```diff
@@ -37,51 +46,64 @@ pub trait Sampleable {
 // Implementations
 //--------------------------------------------------------------------------------------------------
 
-#[async_trait::async_trait(?Send)]
-impl BlockStore for MockStore {
-    #[inline]
-    async fn get_block(&self, cid: &Cid) -> Result<Bytes> {
-        self.0.get_block(cid).await
-    }
+impl MockStore {
+    pub async fn get_block_snapshot(&self, cid: &Cid) -> Result<BlockSnapshot> {
+        let cbor_bytes = self.get_block(cid).await?;
 
-    #[inline]
-    async fn put_block(&self, bytes: impl Into<Bytes>, codec: u64) -> Result<Cid> {
-        self.0.put_block(bytes, codec).await
-    }
+        let ipld = Ipld::decode(DagCborCodec, &mut Cursor::new(cbor_bytes.clone()))?;
+        let mut json_bytes = Vec::new();
+        ipld.encode(DagJsonCodec, &mut json_bytes)?;
 
-    async fn get_deserializable<V: DeserializeOwned>(&self, cid: &Cid) -> Result<V> {
-        let bytes = self.get_block(cid).await?;
-        Ok(serde_json::from_slice(&bytes)?)
+        let value = serde_json::from_slice(&json_bytes)?;
+        Ok(BlockSnapshot {
+            value,
+            cbor: cbor_bytes.to_vec(),
+        })
     }
 
-    async fn put_serializable<V: Serialize>(&self, value: &V) -> Result<Cid> {
-        // We serialize the value to raw CBOR bytes first.
-        let cbor = encode(&ipld_serde::to_ipld(value)?, DagCborCodec)?;
-
-        // Then we deserialize it back to `Ipld`. We can be sure value based on the generated cbor.
-        let value: Ipld = decode(&cbor, DagCborCodec)?;
-
-        let json = encode(
-            &ipld_serde::to_ipld(MockData { value, cbor })?,
-            DagJsonCodec,
-        )?;
-
-        self.put_block(json, CODEC_DAG_JSON).await
+    pub fn get_values(&self) -> Result<Vec<BlockSnapshot>> {
```

Comment:
_Can_ we make this return `BTreeMap<String, BlockSnapshot>` instead of `Vec<BlockSnapshot>`? 
The `String`s would be the encoded CIDs. This way we can follow the links in the snapshot.
(Ideally the snapshot is also topologically sorted, but... that's perhaps asking for a lot for the initial version :sweat_smile: )